tools/Generator.CLI/ConsoleShell.cs

```diff
@@ -0,0 +1,126 @@
+				[Option(ShortName = "v", Description = "Verifies whether the code compiles.", BooleanMode = BooleanMode.Explicit)]
+				bool verifyCompilation = true,
+				[Option(Description = "Specifies if the file should be formatted.", BooleanMode = BooleanMode.Explicit)]
+				bool format = false
+				)
+			{
+				var content = TaskTupleAwaiterGenerator.GenerateContent(arity);
+
+				try
+				{
+					if (verifyCompilation)
+					{
+						Console.WriteLine("Verifying if file can compile.");
+						if (!TestCompiler.Verify(content))
+						{
+							Console.WriteLine("Test compilation failed.");
+							return 1;
+						}
+						Console.WriteLine("Test compilation successful.");
+					}
+
+					if (!System.IO.Directory.Exists(Path.GetDirectoryName(file)))
+					{
+						Console.WriteLine($"Creating directory {Path.GetDirectoryName(file)}.");
+						System.IO.Directory.CreateDirectory(Path.GetDirectoryName(file));
+					}
+
+					try
+					{
+						await SaveContentAsync(file, content, format);
+						return 0;
+					}
+					catch (Exception e)
+					{
+						Console.WriteLine("File write process failed.");
+						Console.WriteLine(e);
+						return 2;
+					}
+				}
+				catch (Exception e)
+				{
+					Console.WriteLine("Test compilation failed.");
+					Console.WriteLine(e);
+					return 1;
+				}
+			}
+
+			private static async Task SaveContentAsync(string file, string content, bool format)
+			{
+				if (format)
+				{
+					Console.WriteLine($"Formatting file.");
+					var ws = new AdhocWorkspace();
+					content = Formatter
+						.Format(CSharpSyntaxTree.ParseText(content, CSharpParseOptions.Default).GetRoot(), ws, ws.Options)
+						.ToFullString();
+				}
+
+				if (System.IO.File.Exists(file))
+				{
+					Console.WriteLine($"Deleting old file at {file}.");
+					System.IO.File.Delete(file);
+				}
+
+				Console.WriteLine($"Writing file {file}.");
+
+				using (var fileStream = new FileStream(file, FileMode.Create, FileAccess.Write, FileShare.Write, 1024, true))
+				{
+					using (var streamWriter = new StreamWriter(fileStream, Encoding.UTF8))
+					{
+						await streamWriter.WriteAsync(content);
+					}
+				}
+
+				Console.WriteLine($"Success.");
+			}
+
+			[ApplicationMetadata(
+				Name = "d",
+				Description = "Generates the file at {directory}\\TaskTupleGenerator.cs.")]
+			public Task<int> Directory(
+				[Argument(Name = "d", Description = "Folder at which the file should be generated.")]
+				string directory,
+				[Option(ShortName = "a", Description = "Highest arity used to generate the file.")]
+				int arity = 16,
+				[Option(ShortName = "v", Description = "Verifies whether the code compiles.", BooleanMode = BooleanMode.Explicit)]
+				bool verifyCompilation = true,
+				[Option(Description = "Specifies if the file should be formatted.", BooleanMode = BooleanMode.Explicit)]
+				bool format = false
+				)
+			{
+				return File(Path.Combine(directory, "TaskTupleExtensions.cs"), arity, verifyCompilation, format);
```

Comment:
Mhm actually when i inspected the csproj's again i dropped down to 2.0 now, to increase range. What would you recommend in terms of specifying targetframeworks? My knowledge is "go for lowest possible" - so i should probably check with which 1.x / .net this compiles as well.

Reply:
I'd probably use the highest framework already required by building the other projects. It guarantees that you can run this project, assuming you can run the other projects. Since binaries for this code aren't shipped, why would we target something lower or have multiple targets?

Reply:
Personally i always find it a hassle being bothered nonstop by every program i install, to update runtimes, which is why i usually prefer lower targets, unless i want to use more current features and have to increase my target version. Both has its ups and downs. So you'd prefer 2.2 then i suppose?

Reply:
So what I'm saying is: Using 2.2 here can't be a hassle because you [already need it](https://github.com/buvinghausen/TaskTupleAwaiter/blob/6c7a7c719800ee8e412f48fa87aac52ac03b5ccf/src/TaskTupleAwaiter.Tests/TaskTupleAwaiter.Tests.csproj#L4) in order to work with the rest of the repository.

Unless you're picturing that there will be folks that will use this tool but won't run open the main solution and run tests?

Reply:
Technically this tool could be converted to be a dotnet global tool as well, where not having the necessity to be on the latest runtime would be an advantage. That's what i'm poking at. If you want me to change it to 2.2 i'll change it.